src/EntryPoints/REST/ApprovePageApi.php

```diff
@@ -33,7 +37,10 @@ public function run( int $pageId ): Response {
 
 		$this->approvalLog->approvePage( $pageId, $this->getAuthority()->getUser()->getId() );
 
-		// TODO: HtmlRepository::saveApprovedHtml( $pageId, $html );
+		$content = $this->pageContentRetriever->getPageContent( $pageId );
+		if ( $content !== null ) {
+			$this->htmlRepository->saveApprovedHtml( $pageId, $content );
+		}
```

Comment:
I'm wondering if `PageContentRetriever` should just return an empty string in the `null` cases, or throw an exception. It does not seem particularly helpful to handle the `null` case.

Specifically, if the page does not exist then something in the code would've broken before this point. And if the ParserOutput is not available, then I'm not sure if the page should not be approved, or if something else should happen.

Reply:
Btw, `getPageContent` is not a good name for a function returning string, since "Content" is a MediaWiki concept.

Reply:
I'll do cleanup in follow-up.